src/services/ApiService.ts

```diff
@@ -200,9 +204,38 @@ export const confirmAccountUpdate = async (params: ConfirmUpdateAccountData): Pr
         .then(statusCheck);
 };
 
+export const getAccountBalance = async (accountId: string) => {
+    const account = await GET(network, `api/v1/accounts/${accountId}`);
+    const tokens = await getAccountTokens(accountId);
+    return {
+        hbars: account.balance.balance / 10 ** 8,
+        tokens
+    };
+}
+
+const getAccountTokens = async (accountId: string) => {
+    const result = [];
+    let nextPage = `api/v1/accounts/${accountId}/tokens`;
+    while (nextPage != null) {
+        const response = await GET(network, nextPage);
+        nextPage = response.links.next ?? null;
+
+        for (const token of response.tokens) {
+            const tokenInfo = await requestTokenInfo(network, token.token_id);
+            result.push({
+                tokenId: token.token_id,
+                balance: token.balance / 10 ** tokenInfo.decimals,
+            });
+        }
+    }
+    return result;
+};
 
 export const requestTokenInfo = async (network: Network, tokenId: string) => {
-    return GET(network,`api/v1/tokens/${tokenId}`);
+    if (!tokenInfoCache[tokenId]) {
```

Comment:
What if there will be an overlap in cache for Testnet and Maine tokens?

Reply:
hardly but possible. fixed